ipatests/test_integration/test_installation_client.py

```diff
@@ -203,3 +267,214 @@ def test_client_nsupdate(self, setup_bindserver):
+            extra_args=['--hostname', 'nonexistent'],
+            raiseonerr=False
+        )
+        exp_msg = "invalid hostname"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_password_no_principal(self):
+        cmd = self.clients[0].run_command(
+            "ipa-client-install --password=random123 -U",
+            raiseonerr=False)
+        exp_msg = "invalid credentials"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_nonexistent_principal(self):
+        cmd = tasks.install_client(
+            self.master, self.clients[0],
+            user="nonexistentprincipal",
+            raiseonerr=False
+        )
+        exp_msg = "not found in Kerberos database"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_non_admin_principal(self):
+        non_admin_user = "testuser"
+        tasks.create_active_user(self.master,
+                                 non_admin_user,
+                                 "RandomPassword123")
+        cmd = tasks.install_client(
+            self.master, self.clients[0],
+            user=non_admin_user,
+            raiseonerr=False
+        )
+        exp_msg = "Password incorrect while getting initial credentials"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_invalid_password(self):
+        cmd = tasks.install_client(
+            self.master, self.clients[0],
+            password="incorrect+password",
+            raiseonerr=False
+        )
+        exp_msg = "Password incorrect while getting initial credentials"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_force_parameter(self):
+        """
+        Second installation should fail except when --force
+        parameter is provided
+        """
+        tasks.install_client(self.master, self.clients[0])
+
+        try:
+            # second installation
+            exp_msg = "IPA client is already configured on this system"
+
+            result = tasks.install_client(self.master,
+                                          self.clients[0],
+                                          raiseonerr=False)
+            assert result.returncode != 0
+            assert exp_msg.lower() in result.stderr_text.lower()
+
+            # second installation, but with force - should still fail
+            result = tasks.install_client(self.master, self.clients[0],
+                                          extra_args=['--force'],
+                                          raiseonerr=False)
+            assert result.returncode != 0
+            assert exp_msg.lower() in result.stderr_text.lower()
+
+            tasks.uninstall_client(self.clients[0], raiseonerr=True)
+
```

Comment:
This is failing because the variable result contains the output from the last failed install. But even if you instead use result = tasks.uninstall_client() the message will still not match because the result will be successful. Did you want to run the uninstaller twice to check this message?

Reply:
Yes, that was a overlook from my side